packages/consumption/src/modules/requests/itemProcessors/readAttribute/ReadAttributeRequestItemProcessor.ts

```diff
@@ -99,19 +232,36 @@ export class ReadAttributeRequestItemProcessor extends GenericRequestItemProcess
     }
 
     public override async applyIncomingResponseItem(
-        responseItem: ReadAttributeAcceptResponseItem | RejectResponseItem,
+        responseItem: ReadAttributeAcceptResponseItem | AttributeSuccessionAcceptResponseItem | RejectResponseItem,
         _requestItem: ReadAttributeRequestItem,
         requestInfo: LocalRequestInfo
-    ): Promise<void> {
-        if (!(responseItem instanceof ReadAttributeAcceptResponseItem)) {
-            return;
+    ): Promise<PeerSharedAttributeSucceededEvent | void> {
+        if (responseItem instanceof ReadAttributeAcceptResponseItem) {
+            await this.consumptionController.attributes.createPeerLocalAttribute({
+                id: responseItem.attributeId,
+                content: responseItem.attribute,
+                peer: requestInfo.peer,
+                requestReference: requestInfo.id
+            });
         }
 
-        await this.consumptionController.attributes.createPeerLocalAttribute({
-            id: responseItem.attributeId,
-            content: responseItem.attribute,
-            peer: requestInfo.peer,
-            requestReference: requestInfo.id
-        });
+        if (responseItem instanceof AttributeSuccessionAcceptResponseItem) {
+            const successorParams = AttributeSuccessorParams.from({
+                id: responseItem.successorId,
+                content: responseItem.successorContent,
+                shareInfo: LocalAttributeShareInfo.from({
+                    peer: requestInfo.peer,
+                    requestReference: requestInfo.id
+                })
+            });
+
+            if (responseItem.successorContent instanceof IdentityAttribute) {
+                const { predecessor, successor } = await this.consumptionController.attributes.succeedPeerSharedIdentityAttribute(responseItem.predecessorId, successorParams);
+                return new PeerSharedAttributeSucceededEvent(this.currentIdentityAddress.toString(), predecessor, successor);
+            }
+            await this.consumptionController.attributes.succeedThirdPartyOwnedRelationshipAttribute(responseItem.predecessorId, successorParams);
```

Comment:
The RelationshipAttribute in question could also be owned by the Recipient of the Request. Therefore, I'd define a function `succeedPeerSharedThirdPartyRelationshipAttribute` function in the AttributesController.ts instead, that works like the succeedPeerSharedRelationshipAttribute function (so that it is still emphasised in which context the succession is performed) and that can be called here. From the perspective of the Sender of the Request, it does not play a role whether the ThirdPartyRelationshipAttribute is owned by the Recipient of the Request or a third party.

Reply:
The peers of a RelationshipAttribute always have the identical versions of that RelationshipAttribute. Succeeding a RelationshipAttribute always implies sharing the successor with the peer. Thus, a ReadAttributeRequestItem can never be answered with the successor of a RelationshipAttribute that wasn't shared before (except it is a ThirdPartyRelationshipAttribute).

Reply:
I have referred to a ThirdPartyRelationshipAttribute, which has as `shareInfo.sourceAttribute` a RelationshipAttribute from the context of the Relationship between the Recipient and a third party, which is owned by the Recipient. If Attribute succession for the RelationshipAttribute is performed between the Recipient and the third party, the Sender is not initially informed about this. However, the Recipient can send the successor in response to a ReadAttributeRequestItem from the Sender, so that Attribute succession can then be performed on the ThirdPartyRelationshipAttribute (which is owned by the Recipient). So I'm also talking about ThirdPartyRelationshipAttributes, but I wanted to emphasise that from the Sender's perspective, it doesn't matter whether it is owned by the Recipient or the third party, which is why I think it is necessary to rename the function.